src/gtk/webview_webkit2_extension.cpp

```diff
@@ -0,0 +1,426 @@
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+#include "wx/defs.h"
+#include "wx/gtk/webview_webkit2_extension.h"
+#include <webkit2/webkit-web-extension.h>
+#define WEBKIT_DOM_USE_UNSTABLE_API
+#include <webkitdom/WebKitDOMDOMSelection.h>
+#include <webkitdom/WebKitDOMDOMWindowUnstable.h>
+
+static const char introspection_xml[] =
+  "<node>"
+  " <interface name='org.wxwidgets.wxGTK.WebExtension'>"
+  "  <method name='ClearSelection'>"
+  "   <arg type='t' name='page_id' direction='in'/>"
+  "  </method>"
+  "  <method name='DeleteSelection'>"
+  "   <arg type='t' name='page_id' direction='in'/>"
+  "  </method>"
+  "  <method name='GetPageSource'>"
+  "   <arg type='t' name='page_id' direction='in'/>"
+  "   <arg type='s' name='source' direction='out'/>"
+  "  </method>"
+  "  <method name='GetPageText'>"
+  "   <arg type='t' name='page_id' direction='in'/>"
+  "   <arg type='s' name='text' direction='out'/>"
+  "  </method>"
+  "  <method name='GetSelectedSource'>"
+  "   <arg type='t' name='page_id' direction='in'/>"
+  "   <arg type='s' name='source' direction='out'/>"
+  "  </method>"
+  "  <method name='GetSelectedText'>"
+  "   <arg type='t' name='page_id' direction='in'/>"
+  "   <arg type='s' name='text' direction='out'/>"
+  "  </method>"
+  "  <method name='HasSelection'>"
+  "   <arg type='t' name='page_id' direction='in'/>"
+  "   <arg type='b' name='has_selection' direction='out'/>"
+  "  </method>"
+  " </interface>"
+  "</node>";
+
+class wxWebViewWebKitExtension;
+
+static gboolean
+wxgtk_webview_authorize_authenticated_peer_cb(GDBusAuthObserver *observer,
+                                              GIOStream *stream,
+                                              GCredentials *credentials,
+                                              wxWebViewWebKitExtension *extension);
+static void
+wxgtk_webview_dbus_connection_created_cb(GObject *source_object,
+                                         GAsyncResult *result,
+                                         wxWebViewWebKitExtension *extension);
+
+static wxWebViewWebKitExtension *extension = NULL;
+
+class wxWebViewWebKitExtension
+{
+public:
+    wxWebViewWebKitExtension(WebKitWebExtension *webkit_extension, const char* server_address);
+    void ClearSelection(GVariant *parameters, GDBusMethodInvocation *invocation);
+    void DeleteSelection(GVariant *parameters, GDBusMethodInvocation *invocation);
+    void GetPageSource(GVariant *parameters, GDBusMethodInvocation *invocation);
+    void GetPageText(GVariant *parameters, GDBusMethodInvocation *invocation);
+    void GetSelectedSource(GVariant *parameters, GDBusMethodInvocation *invocation);
+    void GetSelectedText(GVariant *parameters, GDBusMethodInvocation *invocation);
+    void HasSelection(GVariant *parameters, GDBusMethodInvocation *invocation);
+
+    GDBusConnection *m_dbus_connection;
+    WebKitWebExtension *m_webkit_extension;
```

Comment:
m_dbus_connection does.  m_webkit_extension appears that it does not, I'll correct that.